components/feature/readerview/src/main/java/mozilla/components/feature/readerview/internal/ReaderViewConfig.kt

```diff
@@ -0,0 +1,102 @@
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_SET_COLOR_SCHEME
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_SET_FONT_TYPE
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_SHOW
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_VALUE
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_VALUE_SHOW_COLOR_SCHEME
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_VALUE_SHOW_FONT_SIZE
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.ACTION_VALUE_SHOW_FONT_TYPE
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.COLOR_SCHEME_KEY
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_SIZE_DEFAULT
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_SIZE_KEY
+import mozilla.components.feature.readerview.ReaderViewFeature.Companion.FONT_TYPE_KEY
+import org.json.JSONObject
+import java.util.Locale
+
+/**
+ * Class to store the user configuration for reader view.
+ * Values are initialized lazily, and will send a message when they are changed.
+ */
+internal class ReaderViewConfig(
+    private val prefs: SharedPreferences,
+    private val sendConfigMessage: (JSONObject) -> Unit
+) {
+
+    private var colorSchemeCache: ReaderViewFeature.ColorScheme? = null
+    private var fontTypeCache: ReaderViewFeature.FontType? = null
+    private var fontSizeCache: Int? = null
+
+    var colorScheme: ReaderViewFeature.ColorScheme
+        get() {
+            if (colorSchemeCache == null) {
+                colorSchemeCache = getEnum(COLOR_SCHEME_KEY, ReaderViewFeature.ColorScheme.LIGHT)
+            }
+            return colorSchemeCache!!
+        }
+        set(value) {
+            colorSchemeCache = value
+            setEnumAndSend(COLOR_SCHEME_KEY, ACTION_SET_COLOR_SCHEME, value)
+        }
+
+    var fontType: ReaderViewFeature.FontType
+        get() {
+            if (fontTypeCache == null) {
+                fontTypeCache = getEnum(FONT_TYPE_KEY, ReaderViewFeature.FontType.SERIF)
+            }
+            return fontTypeCache!!
+        }
+        set(value) {
+            fontTypeCache = value
+            setEnumAndSend(FONT_TYPE_KEY, ACTION_SET_FONT_TYPE, value)
+        }
+
+    var fontSize: Int
+        get() {
+            if (fontSizeCache == null) {
+                fontSizeCache = prefs.getInt(FONT_SIZE_KEY, FONT_SIZE_DEFAULT)
+            }
+            return fontSizeCache!!
+        }
+        set(value) {
+            val diff = value - fontSize
+            fontSizeCache = value
+            prefs.edit().putInt(FONT_SIZE_KEY, value).apply()
+            sendMessage(ACTION_CHANGE_FONT_SIZE) { put(ACTION_VALUE, diff) }
+        }
+
+    fun createShowReaderMessage(): JSONObject {
```

Comment:
See above, let's move this back out. I think it makes more sense for it to live in the feature, together with `createHideReaderMessage`.